exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/EasyEVFBatchReader.java

```diff
@@ -0,0 +1,329 @@
+import org.apache.drill.exec.physical.impl.scan.file.FileScanFramework.FileSchemaNegotiator;
+import org.apache.drill.exec.physical.impl.scan.framework.ManagedReader;
+import org.apache.drill.exec.physical.resultSet.ResultSetLoader;
+import org.apache.drill.exec.physical.resultSet.RowSetLoader;
+import org.apache.drill.exec.record.metadata.ColumnMetadata;
+import org.apache.drill.exec.record.metadata.MetadataUtils;
+import org.apache.drill.exec.vector.accessor.ScalarWriter;
+import org.apache.drill.exec.vector.accessor.TupleWriter;
+import org.apache.hadoop.mapred.FileSplit;
+import org.joda.time.Instant;
+import org.joda.time.LocalDate;
+import org.joda.time.Period;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+
+
+/**
+ * To create a format plugin, there is often a great deal of cut/pasted code. The EasyEVFBatchReader
+ * is intended to allow developers of new format plugins to focus their energy on code that is truly unique
+ * for each format.
+ * <p>
+ * To create a new format plugin, simply extend this class, and overwrite the open() method as shown below in the
+ * snippet below. The code that is unique for the formats will be contained in the iterator class which
+ * <p>
+ * With respect to schema creation, there are three basic situations:
+ * <ol>
+ *   <li>Schema is known before opening the file</li>
+ *   <li>Schema is known (and unchanging) after reading the first row of data</li>
+ *   <li>Schmea is completely flexible, IE: not consistent and not known after first row.</li>
+ * </ol>
+ *
+ * This class inmplements a series of methods to facilitate schema creation. However, to achieve the best
+ * possible performance, it is vital to use the correct methods to map the schema. Drill will perform fastest when
+ * the schema is known in advance and the writers can be stored in a data structure which minimizes the amount of string
+ * comparisons.
+ *
+ * <b>Current state</b>
+ * For the third scenario, where the schema is not consistent, there are a collection of functions, all named <pre>writeDataType()</pre>
+ * which accept a ScalarWriter, column name and value. These functions first check to see whether the column has been added to the schema, if not
+ * it adds it. If it has been added, the value will be added to the column.
+ *
+ * <p>
+ *
+ *   <pre>
+ *   public boolean open(FileSchemaNegotiator negotiator) {
+ *     super.open(negotiator);
+ *     super.fileIterator = new LTSVRecordIterator(getRowWriter(), reader);
+ *     return true;
+ *   }
+ *   </pre>
+ *
+ */
+public abstract class EasyEVFBatchReader implements ManagedReader<FileSchemaNegotiator> {
+
+  private static final Logger logger = LoggerFactory.getLogger(EasyEVFBatchReader.class);
+
+  public FileSplit split;
+
+  public EasyEVFIterator fileIterator;
```

Comment:
The file iterator really only applies to the limited set of record-based files. For things like the JSON reader we'd end up ignoring this iterator which might make things a bit more complex.

Reply:
Correct.  I imagined these "Easy" classes to be used for relatively easy formats or for formats where there is some pre-existing java parser.  Thus the design pattern would basically be:
1.  Instantiate the parser, 
2.  Wrap in the easy iterator 
3.  Map fields to cols. 
etc....

This could not be used for JSON, HDF5 or anything more complicated like that.